src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -238,6 +238,7 @@ const char* ClassPathEntry::copy_path(const char* path) {
 }
 
 ClassFileStream* ClassPathDirEntry::open_stream(JavaThread* current, const char* name) {
+  assert(current == Thread::current(), "current is not equal to Thread::current()");
```

Comment:
I understand the defensive programming philosophy, but the whole point of naming the parameter `current` is because it _is_ the current thread. We did a lot of cleanup of these JavaThread parameters (use of THREAD, CHECK etc) to make such things clear and to avoid the need to assert in every call that expects to be passed the current thread. If an API is called externally an assert may be warranted, but otherwise you would put such asserts in every single method that takes the current thread as a parameter and that would be ridiculously redundant. If a thread other than the current thread ever managed to sneak in here it would crash pretty quickly.